app/services/app/app.ts

```diff
@@ -144,14 +146,43 @@ export class AppService extends StatefulService<IAppState> {
     }, 300);
   }
 
-  startLoading() {
-    this.START_LOADING();
-  }
-
-  finishLoading() {
+  /**
+   * Show loading, block the nav-buttons and disable autosaving
+   * If called several times - unlock the screen only after the last function/promise has been finished
+   * Should be called for any scene-collections loading operations
+   * @see RunInLoadingMode decorator
+   */
+  async runInLoadingMode(fn: () => Promise<any> | void) {
+
+    if (!this.state.loading) {
+      this.START_LOADING();
+      this.windowsService.closeChildWindow();
+      this.windowsService.closeAllOneOffs();
+      this.sceneCollectionsService.disableAutoSave();
+    }
+
+    const result = fn();
+    let returningValue = result;
+    if (result instanceof Promise) {
+      const promiseId = uuid();
+      this.loadingPromises[promiseId] = result;
+      returningValue = await result;
+      delete this.loadingPromises[promiseId];
+    }
```

Comment:
We should add some error handling here.  Currently, any exceptions raised during the execution of `fn` (or if it returns a promise that rejects), we will get an infinite loading screen.  The scene collections service is extremely careful to never raise exceptions though its public API, but as this is meant to be more general purpose, we should handle errors in a better way than infinite loading.

Reply:
good catch